packages/babel-traverse/src/path/index.js

```diff
@@ -67,24 +67,16 @@ export default class NodePath {
 
     const targetNode = container[key];
 
-    const paths = pathCache.get(parent) || [];
-    if (!pathCache.has(parent)) {
+    let paths = pathCache.get(parent);
+    if (!paths) {
+      paths = new Map();
```

Comment:
We can't use a `WeakMap` here because we iterate on pathCache in https://github.com/babel/babel/pull/12302/files#diff-e406da86b11f8d4846c6ffb3bf527e1c0fa230888127f90bb8978c16a9bd025aR166

Reply:
We can create an `IterableWeakMap`, falling back to a normal `Map` on Node.js <13

https://github.com/nodejs/node/pull/35915

Reply:
Oh the `IterableWeakMap` is not a builtin object nor supposed to be a public Node.js API, I suggest we hold off a bit since `WeakRef` (relied by `IterableWeakMap`) is still stage 3 and `@babel/traverse` is heavily used.